doc/spec/support-assessment.md

```diff
@@ -0,0 +1,237 @@
+# Support Assessment Dashboard: Infrastructure
+
+## Problem Description
+
+The support assessment dashboard is a feature giving end users insight
+into how well their cluster conforms to a support envelope based on
+continuously measuring and evaluating system state. This document
+proposes infrastructure to effectively do this.
+
+We can roughly divide the problem into:
+1. Data sources measuring system state
+2. Data gathering, preparation, and joining multiple sources
+3. Rule evaluation
+4. User interface (alerts, notifications, dashboard)
+
+The document focuses on 2. 3. while touching 1.
+
+## Challenges
+
+- When gathering and joining data we must take into account that
+  keeping a copy of all state in memory won't scale
+- We must join data from diverse sources like the manger, Prometheus,
+  cluster maps, Aquarium, Ceph orchestrator. All having different schema.
+- Rules are subject to configuration and change. Changes here must be easy.
+
+## Proposed Change
+
+Use Prometheus to gather, prepare, join and evaluate rules on numeric
```

Comment:
My question is - if we convert everything to a metric Prometheus can interact with, does this mean we lose fidelity for reporting issues to users later, since we may no longer have the original datum to report?

Reply:
Sometimes we do, yes. I expect this mostly in cases where we have to hide complex or string metric processing behind gauges/counters. To get more detail when something fires we could add another endpoint to the exporters to be fetched when a user hits the dashboard. 

To make it more concrete an Alertmanager rule could look like this:
```
  - alert: OutdatedClients
    expr: outdated_clients > 0
    for: 23m
    labels:
      severity: warning
    annotations:
      summary: Cluster has {{ $value }} outdated clients
      details: https://@@VERSION_EXPORTER@@/details
```

(Where @@VERSION_EXPORTER@@ is something we resolve from our cluster state)

I've seen people add *dashboard* annotations pointing to things like Grafana. This kind of goes in that direction.